src/language-js/utils.js

```diff
@@ -8,7 +8,15 @@ function hasFlowShorthandAnnotationComment(node) {
     node.extra &&
     node.extra.parenthesized &&
     node.trailingComments &&
-    node.trailingComments[0].value.trimLeft().startsWith(":")
+    // We intentionally only match spaces and tabs here instead of any whitespace because
+    // Flow annotation comments cannot be split across lines. For example:
+    //
+    // (this /*
+    // : any */).foo = 5;
+    //
+    // is not picked up by Flow, so removing the newline would create a type annotation
+    // that the user did not intend to create.
+    node.trailingComments[0].value.match(/^[ \t]*:/)
```

Comment:
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) `\s` is equivalent to:

```
 \f\n\r\t\v\u00a0\u1680\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff
```

I just tried it, and Flow allows any of those except the JS line terminators `\r`, `\n`, `\u2028` and `\u2029`.

This regex might do the trick: `/^(?:(?=.)\s)*:/`